### PR TITLE
feat: added port config for prometheus metrics

### PR DESCRIPTION
**Pull request by Proto007**
## What does this PR address?

Users can configure a port for prometheus metrics using `config.yml`. Works for both grpc and uvicorn server. For uvicorn server, if the specified port is busy, users can still access the metrics using `/metrics` endpoint.
 
## Before submitting:

- [X] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [X] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [X] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?
@aarnphm @ssheng @sauyon  Any feedback is appreciated as always!

### CODE REVIEW

File: bentoml/_internal/configuration/default_configuration.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ api_server:
   metrics:
     enabled: True
     namespace: BENTOML
+    port: 9090
```
[the remark this answers is not in the document]
Comment (Proto007) [reply bentoml/_internal/configuration/default_configuration.yaml:11]: I thought we were using 9090 as default?

File: bentoml/_internal/server/grpc_app.py [hunk 2]
```diff
@@ -90,15 +93,19 @@ def __call__(self) -> GRPCServer:
...
-            self._metrics_client.start_http_server(port)
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                if sock.connect_ex(("localhost", self._metrics_port)) != 0:
+                    self._metrics_client.start_http_server(self._metrics_port)
+                    logger.info(
+                        f"Prometheus metrics for grpc server can be accessed at http://127.0.0.1:{self._metrics_port}"
+                    )
+                else:
+                    logger.warning(
+                        f"Port {self._metrics_port} is already in use. Try using a different port."
```
[the remark this answers is not in the document]
Comment (Proto007) [reply bentoml/_internal/server/grpc_app.py:107]: `self._metrics_client.start_http_server(self._metrics_port)` does not throw any errors if the port is already in use.

**Review by bojiang — CHANGES_REQUESTED** [review]
Shall we have another section `grpc` in the comfig?

**Comment by Proto007** [comment]
@bojiang I believe we have it already. Are you suggesting we have a separate `metrics_port` section under `grpc`?https://github.com/bentoml/BentoML/blob/258f5ec91c8fa5079dca7c83708d53a3caa3aa45/bentoml/_internal/configuration/default_configuration.yaml#L26-L28

**Comment by bojiang** [comment]
@Proto007 sounds great. Plus `port` config always come with `host`. So let's add an option for listening host as well?

**Review by aarnphm — APPROVED** [review]
Good job 🥳

**Pull request merged**